### PR TITLE
mgr: fix MgrStandby eating messages

### PR DESCRIPTION
**Pull request by jcsp**
It was failing to pass up the Mgr::ms_dispatch
return value, so messages were being consumed
even if Mgr was returning false.  Mgr returns
false to enable Client to see FSMap, so the
symptom was the Client not working.

While we're here, refactor switch into if, because
there were only two paths in the switch and we
don't expect to be adding more.

Signed-off-by: John Spray <john.spray@redhat.com>

### CODE REVIEW

File: src/mgr/MgrStandby.cc [hunk 1]
```diff
@@ -304,23 +304,18 @@ bool MgrStandby::ms_dispatch(Message *m)
...
-      }
+  if (m->get_type() == MSG_MGR_MAP) {
+    handle_mgr_map(static_cast<MMgrMap*>(m));
+    return true;
+  } else if (active_mgr) {
+      auto am = active_mgr;
+      lock.Unlock();
+      bool handled = am->ms_dispatch(m);
+      lock.Lock();
+      return handled;
```
Comment (liewegas) [inline src/mgr/MgrStandby.cc]: indentation?

Comment (jcsp) [inline src/mgr/MgrStandby.cc]: Fixed -- in my defense, I wrote it in the back of a car :-)

**Comment by tchaikov** [comment]
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  failed to decode
terminate called after throwing an instance of 'std::runtime_error'
  what():  failed to decode
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/encoding/readable.sh: line 104:  4162 Aborted                 (core dumped) $CEPH_DENCODER type $type import $vdir/objects/$type/$f decode dump_json > $tmp1
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/encoding/readable.sh: line 104:  4163 Aborted                 (core dumped) $CEPH_DENCODER type $type import $vdir/objects/$type/$f decode encode decode dump_json > $tmp2
**** failed to decode /home/jenkins-build/build/workspace/ceph-pull-requests/ceph-object-corpus/archive/0.80-rc1-35-g4812150/objects/MMDSSlaveRequest/e741e9175f555c9e7176f59137edbfac ****
```

retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-16_18:45:23-rados-wip-sage-testing-distro-basic-smithi/

was a bit inconclusive about mgr stat updates.. going to run this through qa again!

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-17_13:41:40-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**